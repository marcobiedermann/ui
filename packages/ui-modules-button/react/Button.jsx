import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import styles from '../button.css';

const Button = ({ children, disabled }) => (
  <button
    className={classNames(styles.button)}
    type="button"
    disabled={disabled}
  >
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.node,
  disabled: PropTypes.bool,
};

Button.defaultProps = {
  children: null,
  disabled: false,
};

export default Button;
