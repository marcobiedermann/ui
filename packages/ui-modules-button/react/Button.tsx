import classNames from 'classnames';
import React, { FC, ButtonHTMLAttributes } from 'react';
import styles from '../button.css';

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {}

const Button: FC<ButtonProps> = (props) => {
  const { className, ...otherProps } = props;

  return <button className={classNames(className, styles.button)} {...otherProps} />;
};

export default Button;
