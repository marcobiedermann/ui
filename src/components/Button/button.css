:root {
  --button-background-color: var(--color-primary);
  --button-color: #fff;
  --button-disabled-background-color: var(--color-info);
}

button {
  background-color: transparent;
  border: 0;
  color: inherit;
  cursor: pointer;
  font: inherit;
  margin: 0;
  outline: 0;
  padding: 0;
}

.button {
  background-color: var(--button-background-color);
  border-radius: 999em;
  color: var(--button-color);
  display: inline-block;
  font-size: 0.875rem;
  min-width: 8em;
  padding: 0.5em 1.5em;
  transition: background-color 0.3s;
}

.button:focus,
.button:hover {
  background-color: #2c387d;
}

.button[disabled] {
  background-color: var(--button-disabled-background-color);
  cursor: not-allowed;
}
