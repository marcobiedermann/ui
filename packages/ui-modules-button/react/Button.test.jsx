import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Button>
          Button
        </Button>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled correctly', () => {
    const tree = renderer
      .create(
        <Button disabled>
          Button
        </Button>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
