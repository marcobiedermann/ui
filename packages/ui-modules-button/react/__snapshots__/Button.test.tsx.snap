// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<button
  className="button"
  type="button"
>
  Button
</button>
`;

exports[`Button component renders disabled correctly 1`] = `
<button
  className="button"
  disabled={true}
  type="button"
>
  Button
</button>
`;
